Bug Fixes

Alex-developer opened the pull request:
Eric

This PR provides fixes for

https://github.com/thomasjacquin/allsky/issues/2858 - [BUG]: Unnecessary 15px bottom margin on overlay editor
https://github.com/thomasjacquin/allsky/issues/2860 - [BUG] Text alignment is slightly off when comparing the overlay editor to the final image
https://github.com/thomasjacquin/allsky/issues/2856 - [BUG] Missing ERROR message for text fields totally outside of image dimensions
https://github.com/thomasjacquin/allsky/issues/2855 - [BUG]: "ERROR: ${MEAN2} has no variable type; check 'fields.json'. Using '???' instead." refers to wrong file
https://github.com/thomasjacquin/allsky/issues/2854 - [BUG] Overlay issue - Moon altitude not correct
https://github.com/thomasjacquin/allsky/issues/2851 - Hide temperature variables for Rpi cameras until the kernel module is fixed
https://github.com/thomasjacquin/allsky/issues/2838 - [BUG]: Default stroke color doesn't work
https://github.com/thomasjacquin/allsky/issues/2837 - [BUG]: Changing field contents or font size resets X to 0
https://github.com/thomasjacquin/allsky/issues/2833 - [BUG] Ensure overlay processor only processes .json and .txt files
https://github.com/thomasjacquin/allsky/issues/2861 - [BUG] Fixes for the overlay editor

Can you cast your eyes over it and if you are happy I'll merge it. It will require a fair bit of testing as there are some quite fundamental changes in the way overlay fields are positioned.

EricClaeys reviewed (approved):
As we discussed in Discord, excludeVariables will need rework to take into account `hasSensorTemperature`

Alex-developer commented:
> As we discussed in Discord, excludeVariables will need rework to take into account `hasSensorTemperature`

Eric

Updated the code to use the hasSensorTemperature. I fixed the error in the ASICameraInfoArray array locally only so that will need doing, I wasnt sure of you are doing anything to the capture routines

EricClaeys reviewed (approved):
Thanks for making the changes.
